### PR TITLE
Automatically deploy doxygen documentation on master branch pushes via travis

### PR DESCRIPTION
**Pull request by initramfs**
This pull request automatically adds the ability for travis to build doxygen documentation from the qTox sources when being run on the master branch and publishes it to the [doxygen project](https://github.com/qTox/doxygen) under the [qTox organization](https://github.com/qTox).

Every time master is changed, it's documentation is automatically updated by travis and can be accessed from [https://qtox.github.io/doxygen/](https://qtox.github.io/doxygen/).

---

This change is [Reviewable](https://reviewable.io/reviews/qtox/qtox/3614)

### CODE REVIEW

**Comment by zetok** [comment]
Review status: 0 of 2 files reviewed at latest revision, 5 unresolved discussions.

---

_[.travis.yml, line 22 [r1]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOpJTjbSg65r-L3jkp3:-KOpJTjco1iXq49or1sh:b-b1owwl) ([raw file](https://github.com/qtox/qtox/blob/d9db152f6f4585a3f260c62c0c478311454760b5/.travis.yml#L22)):_

> ``` YAML
> 
> after_success:
> - test $TRAVIS_PULL_REQUEST == "false" && test $TRAVIS_BRANCH == "master" && bash ./.travis/build-docs.sh
> ```

After successful build, docs should be deployed. Currently though, docs are being build(!) and deployed in `after_success`.

Since currently docs produce some warnings, they can't be made (yet) to fail on warnings, but as soon as warnings on master will be fixed, building docs should be required to fail if there are any warnings. And that can't be done in `after_success`.

The script probably should be split, or something, since it doesn't just build docs, but builds & deploys them.

---

_[.travis/build-docs.sh, line 18 [r1]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOpTLDhC48SDD2Al4_u:-KOpTLDhC48SDD2Al4_v:b-g6d1fl) ([raw file](https://github.com/qtox/qtox/blob/d9db152f6f4585a3f260c62c0c478311454760b5/.travis/build-docs.sh#L18)):_

> ``` Shell
> #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
> ```

It would be the best to set `set -eu -o pipefail` for any new script, but depending on how things work, some things might need to be disabled. (most likely `-u`)

---

_[.travis/build-docs.sh, line 20 [r1]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOpSjJXg8-eDdrK5y-s:-KOpSjJXg8-eDdrK5y-t:b-a5lq7t) ([raw file](https://github.com/qtox/qtox/blob/d9db152f6f4585a3f260c62c0c478311454760b5/.travis/build-docs.sh#L20)):_

> ``` Shell
> 
> # Obtain doxygen
> sudo apt-get install doxygen
> ```

This doesn't fit here. Not sure where & how it should be obtained on travis, but this feels wrong.

This actually would fit in with refector done by iphy, where there are simple scripts (IIRC) just for installing deps for each OS. There it could be added with a comment "dev dep", or something like that.

---

_[.travis/build-docs.sh, line 29 [r1]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOpSC_Bljtjc8cgRTkK:-KOpSC_Bljtjc8cgRTkL:b15nx2r) ([raw file](https://github.com/qtox/qtox/blob/d9db152f6f4585a3f260c62c0c478311454760b5/.travis/build-docs.sh#L29)):_

> ``` Shell
> 
> # Create temporary config file
> cp "$CONFIG_FILE" "$CONFIG_FILE_TMP"
> ```

~All of this is unnecessary – since this is a script that is supposed to be run only on travis, we don't care about modifying the file – it's safe to just append to the `doxygen.conf`.

Unless you want to make `build-docs.sh` script usable also outside travis, in which case it shouldn't be in `.travis/` dir.

---

_[.travis/build-docs.sh, line 49 [r1]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOpTr_IvXBTTqmto8xQ:-KOpTr_IvXBTTqmto8xR:bm8vouz) ([raw file](https://github.com/qtox/qtox/blob/d9db152f6f4585a3f260c62c0c478311454760b5/.travis/build-docs.sh#L49)):_

> ``` Shell
> 
> git add .
> git commit -m "Deploy to GH pages from commit: $GIT_CHASH"
> ```

`--quiet` is needed, since there's quite a lot of files generated & committed.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/3614)_

**Comment by initramfs** [comment]
Review status: 0 of 2 files reviewed at latest revision, 5 unresolved discussions.

---

_[.travis.yml, line 22 [r1]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOpJTjbSg65r-L3jkp3:-KOpWK0p8VLxwnRsC_4D:bqtnjbl) ([raw file](https://github.com/qtox/qtox/blob/d9db152f6f4585a3f260c62c0c478311454760b5/.travis.yml#L22)):_
<details><summary>

<i>Previously, zetok wrote…</i></summary>



> After successful build, docs should be deployed. Currently though, docs are being build(!) and deployed in `after_success`.
> 
> Since currently docs produce some warnings, they can't be made (yet) to fail on warnings, but as soon as warnings on master will be fixed, building docs should be required to fail if there are any warnings. And that can't be done in `after_success`.
> 
> The script probably should be split, or something, since it doesn't just build docs, but builds & deploys them.
> </details>
> Yep, I literally had that thought just now too. I'll split the script in the next commit.

---

_[.travis/build-docs.sh, line 18 [r1]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOpTLDhC48SDD2Al4_u:-KOpWos0O0WsCNPMuvcE:bgzqx9a) ([raw file](https://github.com/qtox/qtox/blob/d9db152f6f4585a3f260c62c0c478311454760b5/.travis/build-docs.sh#L18)):_
<details><summary>

<i>Previously, zetok wrote…</i></summary>



> It would be the best to set `set -eu -o pipefail` for any new script, but depending on how things work, some things might need to be disabled. (most likely `-u`)
> </details>
> Yep okay, will do.

---

_[.travis/build-docs.sh, line 20 [r1]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOpSjJXg8-eDdrK5y-s:-KOpWwmcv-OV_wRH-kjj:bpst1dz) ([raw file](https://github.com/qtox/qtox/blob/d9db152f6f4585a3f260c62c0c478311454760b5/.travis/build-docs.sh#L20)):_
<details><summary>

<i>Previously, zetok wrote…</i></summary>



> This doesn't fit here. Not sure where & how it should be obtained on travis, but this feels wrong.
> 
> This actually would fit in with refector done by iphy, where there are simple scripts (IIRC) just for installing deps for each OS. There it could be added with a comment "dev dep", or something like that.
> 
> </details>
> An idea I just had was, spawn another travis task that's dedicated to build documentation only. We can assign a linux environment with basically only that line. Like this, not only do we make the script neater, we put less stress on travis (since after the implementation of doxygen-fails-the-build), we can fail out builds that violate documentation standards a lot faster.

E.g. User A creates PR with some changes, travis spawns 3 jobs: linux-docs, linux-compile, OSX-compile. Because doxygen only takes seconds to run, should there be an error, travis can immediately note that the PR is failing and cannot be merged (and as maintainers we can inform people on this, perhaps a new label for "pr-doc-change-required"?

---

_[.travis/build-docs.sh, line 29 [r1]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOpSC_Bljtjc8cgRTkK:-KOpYpjIC77LB6kP-C31:bv19xqd) ([raw file](https://github.com/qtox/qtox/blob/d9db152f6f4585a3f260c62c0c478311454760b5/.travis/build-docs.sh#L29)):_
<details><summary>

<i>Previously, zetok wrote…</i></summary>



> ~All of this is unnecessary – since this is a script that is supposed to be run only on travis, we don't care about modifying the file – it's safe to just append to the `doxygen.conf`.
> 
> Unless you want to make `build-docs.sh` script usable also outside travis, in which case it shouldn't be in `.travis/` dir.
> </details>
> Ok. Will fix, most of this was just copied from my own script, so it was rather defensive.

---

_[.travis/build-docs.sh, line 49 [r1]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOpTr_IvXBTTqmto8xQ:-KOpYzojwro9MTNdorfP:b-rums5c) ([raw file](https://github.com/qtox/qtox/blob/d9db152f6f4585a3f260c62c0c478311454760b5/.travis/build-docs.sh#L49)):_
<details><summary>

<i>Previously, zetok wrote…</i></summary>



> `--quiet` is needed, since there's quite a lot of files generated & committed.
> </details>
> Okay. Will do.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/3614)_

**Comment by zetok** [comment]
Review status: 0 of 2 files reviewed at latest revision, 5 unresolved discussions.

---

_[.travis/build-docs.sh, line 20 [r1]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOpSjJXg8-eDdrK5y-s:-KOpZuEuc7JdgbfW6PLP:b-hje19h) ([raw file](https://github.com/qtox/qtox/blob/d9db152f6f4585a3f260c62c0c478311454760b5/.travis/build-docs.sh#L20)):_
Separate travis instance would do. While at it, verifying commit messages also could be moved to it.

> Because doxygen only takes seconds to run, should there be an error, travis can immediately note that the PR is failing and cannot be merged (…)

Yep. https://docs.travis-ci.com/user/customizing-the-build/#Fast-Finishing

> (…) and as maintainers we can inform people on this, perhaps a new label for "pr-doc-change-required"?

No need for a new label IMO, since it should rather clear why PR failed. What could come useful though, is some documentation on what is checked on CI, so that people would know for what to look out.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/3614)_

**Comment by zetok** [comment]
Reviewed 2 of 6 files at r2, 4 of 6 files at r3.
Review status: 5 of 7 files reviewed at latest revision, 4 unresolved discussions.

---

_[.travis/build-docs.sh, line 19 [r2]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOsC1zyLZ-zsmKd1EO2:-KOsC1zyLZ-zsmKd1EO3:b-bqikh9) ([raw file](https://github.com/qtox/qtox/blob/14f977df42b1e48ac89a134a6f0c16d26ab8bbaa/.travis/build-docs.sh#L19)):_

> ``` Shell
> 
> # Fail out on error
> set -e -o pipefail
> ```

No `-u`? But it doesn't look like removing it is needed?

---

_[.travis/deploy-docs.sh, line 24 [r3]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOsEbxOTBfFN3eo8lUB:-KOsEbxOTBfFN3eo8lUC:b-munjid) ([raw file](https://github.com/qtox/qtox/blob/504905aff188bd3e824eef956ba71f8e928146f7/.travis/deploy-docs.sh#L24)):_

> ``` Shell
> if [ ! -d ./doc/html ]
> then
>   exit 1
> ```

Use spaces, not tabs.

Also, you might want to `echo "Failing, since there's no $DOCS_DIR` before `exit 1`.

---

_[.travis/deploy-docs.sh, line 31 [r3]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOsIIWhCKVxVwyHJjt0:-KOsIIWifLOCNS-qFNt3:b-2w56ox) ([raw file](https://github.com/qtox/qtox/blob/504905aff188bd3e824eef956ba71f8e928146f7/.travis/deploy-docs.sh#L31)):_

> ``` Shell
> 
> # Push generated doxygen to GitHub pages
> cd ./doc/html/
> ```

can be a var, also used above

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/3614)_

**Comment by initramfs** [comment]
Review status: 5 of 7 files reviewed at latest revision, 3 unresolved discussions.

---

_[.travis/build-docs.sh, line 19 [r2]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOsC1zyLZ-zsmKd1EO2:-KOsLBW4M12xeOglGMTq:b-ofdhph) ([raw file](https://github.com/qtox/qtox/blob/14f977df42b1e48ac89a134a6f0c16d26ab8bbaa/.travis/build-docs.sh#L19)):_
<details><summary>

<i>Previously, zetok wrote…</i></summary>



> No `-u`? But it doesn't look like removing it is needed?
> </details>
> I'm gonna leave it in just incase someone in the future tweaks this to (accidentally) add unset variables.

---

_[.travis/deploy-docs.sh, line 24 [r3]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOsEbxOTBfFN3eo8lUB:-KOsIvSu-Qy2tLyzjjoE:b-19q78h) ([raw file](https://github.com/qtox/qtox/blob/504905aff188bd3e824eef956ba71f8e928146f7/.travis/deploy-docs.sh#L24)):_
<details><summary>

<i>Previously, zetok wrote…</i></summary>



> Use spaces, not tabs.
> 
> Also, you might want to `echo "Failing, since there's no $DOCS_DIR` before `exit 1`.
> 
> </details>
> I did that before, but since this generates a message on every job (since this is run on every job), I thought to just silently fail out (people might think it's an error). This is run on `after_success` anyway, so if the doc building job failed, this won't even run.

Basically, the only way this check can fail that requires user-intervention is if doxygen produces absolutely no output but exits with code 0 (causing the build-doc job to look like it succeeded).

The tab/space thing is fixed, stupid vim...

---

_[.travis/deploy-docs.sh, line 31 [r3]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOsIIWhCKVxVwyHJjt0:-KOsJa2np6c0pCoBJP03:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/504905aff188bd3e824eef956ba71f8e928146f7/.travis/deploy-docs.sh#L31)):_
<details><summary>

<i>Previously, zetok wrote…</i></summary>



> can be a var, also used above
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/3614)_

**Comment by zetok** [comment]
Review status: 5 of 7 files reviewed at latest revision, 4 unresolved discussions.

---

_[.travis.yml, line 31 [r4]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOsRd-aer14aji9I-KV:-KOsRd-aer14aji9I-KW:b-bl0o23) ([raw file](https://github.com/qtox/qtox/blob/3a437b3cdbaa0138008e808c82c2716b40f627f6/.travis.yml#L31)):_

> ``` YAML
> 
> after_success:
>   - test $TRAVIS_PULL_REQUEST == "false" && test $TRAVIS_BRANCH == "master" && bash ./.travis/deploy-docs.sh
> ```

Should also check whether `$JOB == "build-docs"`

---

_[.travis/build-docs.sh, line 19 [r2]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOsC1zyLZ-zsmKd1EO2:-KOsOOmuae6obOp_nrVT:b-e4rl30) ([raw file](https://github.com/qtox/qtox/blob/14f977df42b1e48ac89a134a6f0c16d26ab8bbaa/.travis/build-docs.sh#L19)):_
<details><summary>

<i>Previously, initramfs wrote…</i></summary>



> I'm gonna leave it in just incase someone in the future tweaks this to (accidentally) add unset variables.
> </details>
> Yeah, no. There should be no accidentally added unset vars, and there will have to be a good reason to remove `-u`.

---

_[.travis/deploy-docs.sh, line 24 [r3]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOsEbxOTBfFN3eo8lUB:-KOsPfSxar_33zTFtcHg:b-2bnvk8) ([raw file](https://github.com/qtox/qtox/blob/504905aff188bd3e824eef956ba71f8e928146f7/.travis/deploy-docs.sh#L24)):_

> I did that before, but since this generates a message on every job (since this is run on every job) (…)

Err?

This shouldn't be executed on every job, just on pushes to `master`. And it does seem to be guarded in `.travis.yml` against running on something other than `master`.

> (…) I thought to just silently fail out (people might think it's an error) (…)

This _is_ an error. And silently failing on error is not a good thing. Yeah, it does happen in "just" one section, but having an error message will be helpful anyway if one day things will break.

> This is run on after_success anyway, so if the doc building job failed, this won't even run.

Yes and no. Docs might build perfectly fine, but on the other hand this script can just fail. I.e. since this script doesn't take doxygen's config values for location of docs. Thus upon change doxygen can happily build docs in a new location, while this would horribly fail when looking for them in the location.

The "solution" would be to just use contents of `doxygen.config` for path where docs should be looked for. Parsing would be tad ugly, but...

---

_[.travis/deploy-docs.sh, line 21 [r4]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOsSItvqsUuqE4vzBWL:-KOsSItwTtfoWmB-Rbca:b-w01uj0) ([raw file](https://github.com/qtox/qtox/blob/3a437b3cdbaa0138008e808c82c2716b40f627f6/.travis/deploy-docs.sh#L21)):_

> ``` Shell
> set -eu -o pipefail
> 
> DOCS_FOLDER="./doc/html"
> ```

`s/FOLDER/DIR/`

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/3614)_

**Comment by initramfs** [comment]
Review status: 4 of 7 files reviewed at latest revision, 4 unresolved discussions.

---

_[.travis.yml, line 31 [r4]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOsRd-aer14aji9I-KV:-KOsVjIZKFD_uUERFzDN:bpewupi) ([raw file](https://github.com/qtox/qtox/blob/3a437b3cdbaa0138008e808c82c2716b40f627f6/.travis.yml#L31)):_
<details><summary>

<i>Previously, zetok wrote…</i></summary>



> Should also check whether `$JOB == "build-docs"`
> </details>
> Ok. Well, I feel kinda stupid now... I've also added the error message back, since now it's meaningful.

---

_[.travis/build-docs.sh, line 19 [r2]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOsC1zyLZ-zsmKd1EO2:-KOsWj-AKIUod2F_UJmp:b-kw2vne) ([raw file](https://github.com/qtox/qtox/blob/14f977df42b1e48ac89a134a6f0c16d26ab8bbaa/.travis/build-docs.sh#L19)):_
<details><summary>

<i>Previously, zetok wrote…</i></summary>



> Yeah, no. There should be no accidentally added unset vars, and there will have to be a good reason to remove `-u`.
> </details>
> Ok. Sure.

---

_[.travis/deploy-docs.sh, line 24 [r3]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOsEbxOTBfFN3eo8lUB:-KOsW7gHhrOQu526SbBF:b-uz9spj) ([raw file](https://github.com/qtox/qtox/blob/504905aff188bd3e824eef956ba71f8e928146f7/.travis/deploy-docs.sh#L24)):_
<details><summary>

<i>Previously, zetok wrote…</i></summary>



> > I did that before, but since this generates a message on every job (since this is run on every job) (…)
> 
> Err?
> 
> This shouldn't be executed on every job, just on pushes to `master`. And it does seem to be guarded in `.travis.yml` against running on something other than `master`.
> 
> > (…) I thought to just silently fail out (people might think it's an error) (…)
> 
> This _is_ an error. And silently failing on error is not a good thing. Yeah, it does happen in "just" one section, but having an error message will be helpful anyway if one day things will break.
> 
> > This is run on after_success anyway, so if the doc building job failed, this won't even run.
> 
> Yes and no. Docs might build perfectly fine, but on the other hand this script can just fail. I.e. since this script doesn't take doxygen's config values for location of docs. Thus upon change doxygen can happily build docs in a new location, while this would horribly fail when looking for them in the location.
> 
> The "solution" would be to just use contents of `doxygen.config` for path where docs should be looked for. Parsing would be tad ugly, but...
> </details>
> Sorry terminology might have been confusing, jobs ≠ builds. What I mean is that, there are 4 jobs (after this PR, they are: verify-commit..., build-docs, build-ubuntu... and buid-osx) per build and all of them would have executed the `after_success` build step (in which 3 of them would display this error message, since they don't have the generated docs). See my comment above about me being stupid and not limiting deployment to the docs job.

As such, I'll add that error message back in.

---

_[.travis/deploy-docs.sh, line 21 [r4]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOsSItvqsUuqE4vzBWL:-KOsWi82wW5biMtbGWPp:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/3a437b3cdbaa0138008e808c82c2716b40f627f6/.travis/deploy-docs.sh#L21)):_
<details><summary>

<i>Previously, zetok wrote…</i></summary>



> `s/FOLDER/DIR/`
> </details>
> Done.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/3614)_

**Comment by zetok** [comment]
>   feat(build): deploy doxygen documentation on master branch pushes 
>   feat(build): add an error message on failure to deploy documentation

That's a `chore`

---

Reviewed 1 of 6 files at r3, 1 of 2 files at r5.
Review status: 6 of 7 files reviewed at latest revision, 2 unresolved discussions.

---

_[.travis/deploy-docs.sh, line 21 [r4]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOsSItvqsUuqE4vzBWL:-KOsdDC6lAdIeqRJ9lLD:b-8nlap4) ([raw file](https://github.com/qtox/qtox/blob/3a437b3cdbaa0138008e808c82c2716b40f627f6/.travis/deploy-docs.sh#L21)):_
<details><summary>

<i>Previously, initramfs wrote…</i></summary>



> Done.
> </details>
> Not yet.

---

_[.travis/deploy-docs.sh, line 26 [r5]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOsdPzy8ApqDOl05j_h:-KOsdPzy8ApqDOl05j_i:b-o8qin1) ([raw file](https://github.com/qtox/qtox/blob/e626de170e66a9e137c2504a4480437fa675cc16/.travis/deploy-docs.sh#L26)):_

> ``` Shell
> if [ ! -d "$DOCS_FOLDER" ]
> then
>     echo "Docs deploy failing, no $DOCS_DIR present."
> ```

This would fail, since script doesn't have `DOCS_DIR` var yet.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/3614#-:-KOsdbyWOMIInpJxpCYV:b-edmg6q)_

**Comment by initramfs** [comment]
Review status: 4 of 7 files reviewed at latest revision, 2 unresolved discussions.

---

_[.travis/deploy-docs.sh, line 21 [r4]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOsSItvqsUuqE4vzBWL:-KOt0pKnSHyuIT5IeJTS:b22fx5b) ([raw file](https://github.com/qtox/qtox/blob/3a437b3cdbaa0138008e808c82c2716b40f627f6/.travis/deploy-docs.sh#L21)):_
<details><summary>

<i>Previously, zetok wrote…</i></summary>



> Not yet.
> </details>
> Done. This time for real...

---

_[.travis/deploy-docs.sh, line 26 [r5]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOsdPzy8ApqDOl05j_h:-KOt0nBV54S34q7PfRlV:bt2vr1b) ([raw file](https://github.com/qtox/qtox/blob/e626de170e66a9e137c2504a4480437fa675cc16/.travis/deploy-docs.sh#L26)):_
<details><summary>

<i>Previously, zetok wrote…</i></summary>



> This would fail, since script doesn't have `DOCS_DIR` var yet.
> </details>
> Fixed.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/3614)_

**Comment by zetok** [comment]
Review status: 4 of 7 files reviewed at latest revision, 3 unresolved discussions.

---

_[.travis/deploy-docs.sh, line 23 [r6]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOt404smtEXIdtRkOgo:-KOt404smtEXIdtRkOgp:b-c0w07n) ([raw file](https://github.com/qtox/qtox/blob/c40ddbcc4e8e1b1766282d0489790ab85a874ada/.travis/deploy-docs.sh#L23)):_

> ``` Shell
> # Extract html documentation directory from doxygen configuration
> OUTPUT_DIR_CFG=( $(grep 'OUTPUT_DIRECTORY' "$DOXYGEN_CONFIG_FILE") )
> HTML_OUTPUT_CFG=( $(grep 'HTML_OUTPUT' "$DOXYGEN_CONFIG_FILE") )
> ```

`DOXYGEN_CONFIG_FILE` is not defined anywhere?

---

_[.travis/deploy-docs.sh, line 25 [r6]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOt3FK7qqXpkJtmnRiR:-KOt3FK8Qe4dk1xkJKL2:b-ecfabl) ([raw file](https://github.com/qtox/qtox/blob/c40ddbcc4e8e1b1766282d0489790ab85a874ada/.travis/deploy-docs.sh#L25)):_

> ``` Shell
> HTML_OUTPUT_CFG=( $(grep 'HTML_OUTPUT' "$DOXYGEN_CONFIG_FILE") )
> 
> DOCS_DIR="$(echo "./${OUTPUT_DIR_CFG[2]}/${HTML_OUTPUT_CFG[2]}/" | sed -r -e 's|//+|/|g')"
> ```

`echo` and `sed` are unnecessary, since any amount of multiple slashes after each other is treated by Linux as a single slash.

Simple `DOCS_DIR="./${OUTPUT_DIR_CFG[2]}/${HTML_OUTPUT_CFG[2]}/"` would do.

---

_[.travis/deploy-docs.sh, line 28 [r6]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOt3iOSGQFc_vc7vaAw:-KOt3iOSGQFc_vc7vaAx:b-5zn0j4) ([raw file](https://github.com/qtox/qtox/blob/c40ddbcc4e8e1b1766282d0489790ab85a874ada/.travis/deploy-docs.sh#L28)):_

> ``` Shell
> 
> echo $DOCS_DIR
> exit 0
> ```

Err? Leftover debug?

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/3614)_

**Comment by initramfs** [comment]
Review status: 4 of 7 files reviewed at latest revision, 3 unresolved discussions.

---

_[.travis/deploy-docs.sh, line 23 [r6]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOt404smtEXIdtRkOgo:-KOt53pRs7t64jPWPFXD:b-lzdagd) ([raw file](https://github.com/qtox/qtox/blob/c40ddbcc4e8e1b1766282d0489790ab85a874ada/.travis/deploy-docs.sh#L23)):_
<details><summary>

<i>Previously, zetok wrote…</i></summary>



> `DOXYGEN_CONFIG_FILE` is not defined anywhere?
> </details>
> I've moved the definition to .travis.yml, during the build-docs docs job. It seems like a bad idea to hard-code the name of the config file in different scripts.

---

_[.travis/deploy-docs.sh, line 25 [r6]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOt3FK7qqXpkJtmnRiR:-KOt5LtAEjrrJhgda3rM:b-896fix) ([raw file](https://github.com/qtox/qtox/blob/c40ddbcc4e8e1b1766282d0489790ab85a874ada/.travis/deploy-docs.sh#L25)):_
<details><summary>

<i>Previously, zetok wrote…</i></summary>



> `echo` and `sed` are unnecessary, since any amount of multiple slashes after each other is treated by Linux as a single slash.
> 
> Simple `DOCS_DIR="./${OUTPUT_DIR_CFG[2]}/${HTML_OUTPUT_CFG[2]}/"` would do.
> </details>
> Done.

---

_[.travis/deploy-docs.sh, line 28 [r6]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOt3iOSGQFc_vc7vaAw:-KOt5MRV0N4oQGAcBXHA:bvw3egk) ([raw file](https://github.com/qtox/qtox/blob/c40ddbcc4e8e1b1766282d0489790ab85a874ada/.travis/deploy-docs.sh#L28)):_
<details><summary>

<i>Previously, zetok wrote…</i></summary>



> Err? Leftover debug?
> </details>
> And..... This is why we have Reviewable... Fixed.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/3614)_

**Comment by zetok** [comment]
:lgtm:

---

Reviewed 2 of 3 files at r6, 1 of 1 files at r7.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_[.travis/deploy-docs.sh, line 23 [r6]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOt404smtEXIdtRkOgo:-KOt5_Vjm41wQT8XRWAh:bgq8eca) ([raw file](https://github.com/qtox/qtox/blob/c40ddbcc4e8e1b1766282d0489790ab85a874ada/.travis/deploy-docs.sh#L23)):_
<details><summary>

<i>Previously, initramfs wrote…</i></summary>



> I've moved the definition to .travis.yml, during the build-docs docs job. It seems like a bad idea to hard-code the name of the config file in different scripts.
> </details>
> Oh, right, sorry, I've just figured out how to make reviewable show all the changes in latest revision, not just the file I'm looking at :s

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/3614#-:-KOt8B0ulH5_sURLZxbF:bnfp4nl)_

**Comment by zetok** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

_[.travis/build-osx.sh, line 14 [r7]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOuNAjfDVt4W2g_f6oC:-KOuNAjfDVt4W2g_f6oD:b-lhk64s) ([raw file](https://github.com/qtox/qtox/blob/7dbc95437b6894cb43035e72cf0fcd0f3141a34f/.travis/build-osx.sh#L14)):_

> ``` Shell
> #    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
> #    GNU General Public License for more details.
> #
> ```

part of the license header is missing

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/3614)_

**Comment by initramfs** [comment]
Review status: 2 of 7 files reviewed at latest revision, 1 unresolved discussion.

---

_[.travis/build-osx.sh, line 14 [r7]](https://reviewable.io:443/reviews/qtox/qtox/3614#-KOuNAjfDVt4W2g_f6oC:-KOvDFKLGZ_xOkE1uyDN:bt2vr1b) ([raw file](https://github.com/qtox/qtox/blob/7dbc95437b6894cb43035e72cf0fcd0f3141a34f/.travis/build-osx.sh#L14)):_
<details><summary>

<i>Previously, zetok wrote…</i></summary>



> part of the license header is missing
> </details>
> Fixed.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/3614)_

**Comment by zetok** [comment]
:lgtm:

---

Reviewed 1 of 6 files at r3, 5 of 5 files at r8.
Review status: all files reviewed at latest revision, all discussions resolved.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/3614#-:-KOvEv-o42HyFDpdb5qg:bnfp4nl)_
